src/pure-async.js

```diff
@@ -0,0 +1,330 @@
+/* istanbul ignore file */
+import * as React from 'react'
+import ReactDOM from 'react-dom'
+import * as ReactDOMClient from 'react-dom/client'
+import {
+  getQueriesForElement,
+  prettyDOM,
+  configure as configureDTL,
+} from '@testing-library/dom'
+import {getIsReactActEnvironment, setReactActEnvironment} from './act-compat'
+import {fireEvent} from './fire-event'
+import {getConfig, configure} from './config'
+
+async function act(scope) {
+  const previousActEnvironment = getIsReactActEnvironment()
+  setReactActEnvironment(true)
+  try {
+    // React.act isn't async yet so we need to force it.
+    return await React.act(async () => {
+      scope()
+    })
```

Comment:
pretty sure this should be
```suggestion
    return await React.act(async () => {
      await scope()
    })
```
or
```suggestion
    return await React.act(async () => {
      return scope()
    })
```